src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/SdkAndPackagesDependenciesSnapshotFilter.cs

```diff
@@ -87,12 +87,10 @@ internal class SdkAndPackagesDependenciesSnapshotFilter : DependenciesSnapshotFi
             out bool filterAnyChanges)
         {
             filterAnyChanges = false;
-            if (!dependency.TopLevel || !dependency.Resolved)
-            {
-                return dependency;
-            }
 
-            if (dependency.Flags.Contains(DependencyTreeFlags.PackageNodeFlags))
+            if (dependency.TopLevel && 
+                dependency.Resolved && 
+                dependency.Flags.Contains(DependencyTreeFlags.PackageNodeFlags))
```

Comment:
The side-effects below are a bit icky :(

Reply:
Via the builders? In this case it's the design. The add/remove filters can manipulate the builders, as well as reject the dependency or (in the case of add) modify it. Maybe the name _filter_ is misleading if you think of it as a pure predicate. Not sure what a better name would be.

Reply:
Yes, I meant because filter implies a pure predicate.  Not sure of a better name either.